x-pack/plugins/spaces/server/default_space/default_space_service.test.ts

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import * as Rx from 'rxjs';
+import {
+  DefaultSpaceService,
+  RETRY_SCALE_DURATION,
+  RETRY_DURATION_MAX,
+} from './default_space_service';
+import {
+  ServiceStatusLevels,
+  ServiceStatusLevel,
+  CoreStatus,
+  SavedObjectsRepository,
+  SavedObjectsErrorHelpers,
+} from '../../../../../src/core/server';
+import { coreMock, loggingServiceMock } from 'src/core/server/mocks';
+import { licensingMock } from '../../../licensing/server/mocks';
+import { SpacesLicenseService } from '../../common/licensing';
+import { ILicense } from '../../../licensing/server';
+import { nextTick } from 'test_utils/enzyme_helpers';
+import { first } from 'rxjs/operators';
+
+const advanceRetry = async (initializeCount: number) => {
+  await Promise.resolve();
+  let duration = initializeCount * RETRY_SCALE_DURATION;
+  if (duration > RETRY_DURATION_MAX) {
+    duration = RETRY_DURATION_MAX;
+  }
+  jest.advanceTimersByTime(duration);
+};
+
+interface SetupOpts {
+  elasticsearchStatus: ServiceStatusLevel;
+  savedObjectsStatus: ServiceStatusLevel;
+  license: ILicense;
+}
+const setup = ({ elasticsearchStatus, savedObjectsStatus, license }: SetupOpts) => {
+  const core = coreMock.createSetup();
+  const { status } = core;
+  status.core$ = (new Rx.BehaviorSubject({
+    elasticsearch: {
+      level: elasticsearchStatus,
+      summary: '',
+    },
+    savedObjects: {
+      level: savedObjectsStatus,
+      summary: '',
+    },
+  }) as unknown) as Rx.Observable<CoreStatus>;
+
+  const { savedObjects } = coreMock.createStart();
+  const repository = savedObjects.createInternalRepository() as jest.Mocked<SavedObjectsRepository>;
+  // simulate space not found
+  repository.get.mockRejectedValue(SavedObjectsErrorHelpers.createGenericNotFoundError());
```

Comment:
**note** I don't _love_ that I'm asserting the behavior of a dependency of a sub-module in these tests, but it seemed like the "safest" thing to do for now. The default space service is inherently coupled to the repository given the work it has to do.

I _could_ `jest.mock` the `create_default_space` module, but I didn't love that idea either.

Reply:
I'd normally lean towards mocking the sub-module, but this seems fine for this suite of unit tests. Like you said, it's inherently coupled.